### PR TITLE
Update to micrometer 1.6.1

### LINKED ISSUE #13079: Upgrade Micrometer to 1.6.1
Need to work around a few different issues before moving up to Micrometer 1.6.0 

* Some incubating API methods were dropped in 1.6.0 (pre-register tags on Timer Sample)
* micrometer-metrics/micrometer#2330 -- NullPointerException in JvmMemory.isOldGenPool in 1.6.0 for AdoptOpenJDK
* micrometer-metrics/micrometer#2334 -- IllegalArgumentException for Hibernate

### PR DESCRIPTION
**Pull request by ebullient**
Consolidate/Align hibernate metrics naming
Resolves #13079

### CODE REVIEW

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java [hunk 1]
```diff
@@ -57,20 +57,23 @@ public void setNumber(Long number) {
     @Test
     @Transactional
     public void testMetrics() {
-        assertEquals(0L, getCounterValueOrNull("hibernate-orm.queries.executed",
+        assertEquals(0L, getCounterValueOrNull("hibernate.query.executions",
```
Comment (gsmet) [inline extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java:60]: Why this change? I was the one asking for `-orm`. Hibernate is more than ORM and we might add Hibernate Search metrics at some point.

**Review by gsmet — CHANGES_REQUESTED** [review]

File: extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java (hunk 1 above)
Comment (Sanne) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java:60]: hi @gsmet - I've been discussing this with Erin, and while I explained her the same, she made some excellent points so she persuaded me.

The gist of the discusion is that Micrometer upstream has been using the "wrong" prefix `hibernate.` for a long time, and it's what is being used by many existing applications. So in practice, the metrics of many applications (most of them will be existing) are being collected in the same metrics-storage, and it's important for the radix to be consistent.

I did remind that Quarkus doesn't aim to migrate existing applications, but this is about coexisting with (many) existing applications.

We also wondered if we should have a configuration property, but after thinking about that I think giving up on the prefix correction is fine - it's a ship that sailed long before we had the chance.

I suppose Hibernate Search will be able to add additional metrics under the same, shared radix.

Not sure about Validator; it is unlikely to need such metrics soon? If we want them, I think we can "steal" the `validator.` prefix and make it our own.

Comment (Sanne) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java:60]: adding @yrodiere for awareness and opinions as well..

Comment (gsmet) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java:60]: I don't expect us to have metrics for Validator.

But having the `hibernate.` prefix alone will certainly be confusing once we add search in the mix.

Comment (yrodiere) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java:60]: I agree with Guillaume that this is an annoying change.

On the communication side of things, it's kind of hard to explain what Hibernate Search is if you go around telling people that Hibernate is an ORM. Hibernate is not an ORM, it's a family of project, and organization, whatever you want, but not *just* an ORM. However, this ship has sailed too when it was decided (recently) to name "Hibernate Reactive" that way. We're clearly going in a direction where "Hibernate" is advertised as the name of an ORM. Explaining what that Hibernate Search is not Hibernate ORM is hard, and it'll stay that way, I gave up.

If I understand correctly, we're sticking to a pre-existing prefix for backward compatibility reasons. I suppose there are no ways to define aliases to just work around the problem? I.e. make sure that people retrieving metrics for the prefix 'hibernate.' will continue to get the metrics for `hibernate-orm.`, with a warning? That would seem like a better solution.

Regardless, when we will add metrics for Hibernate Search, we'll try to stick to whatever is the rule in Quarkus. I suspect reusing the `hibernate.` prefix will lead to problems along the way, so I'd rather avoid that.
I think we're supposed to use the extension name, so we'll end up with something like `hibernate-search-orm-elasticsearch.`? I don't have a strong opinion as long as we avoid naming conflicts with future extensions based on Hibernate Search without ORM.

Comment (Sanne) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java:60]: Remember it's a nested nodes structure, so you could still use the `hibernate.` prefix, as long as we manage to disambiguate sub elements.

Specifically, `hibernate.query.executions` is taken, but you could use `hibernate.fulltextquery.executions`, or `hibernate.search.fulltextquery.executions`.

And yes it's annoying, but I also give up.. let's consider that this PR is about better integration with an existing library which did behave like this.

If one of you (both) feel very strongly this is wrong, I propose you follow up with a patch which makes the prefix configurable; if we default to `hibernate-orm` there's a (very remote) possibility that you manage to eventually re-educate the rest of the universe, but before we get there consider you'll be annoying many users who will be forced to figure out what's wrong, find the configuration property, reconfigure all their apps.

Comment (ebullient) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java:60]: I talked to Sanne about this. This change is more in line w/ the rest of the community's usage of hibernate metrics (it matches what the micrometer binder is doing verbatim, which makes it easier to intermix). @Sanne agreed that consistency across hibernate metrics consumers is more important than the distinction of `-orm` in only Quarkus.

Comment (ebullient) [reply extensions/hibernate-orm/deployment/src/test/java/io/quarkus/hibernate/orm/HibernateMetricsTestCase.java:60]: From a metrics perspective, I feel like there is an education argument to be made for orm vs. search -- but given patterns, I'm not sure how important this really is (other than for people that really care, which y'all do).

In micrometer, everything is dot qualified, which then is translated to whatever a meter registry expects (e.g. `.` to `_`) ... from what I understand, trying to disambiguate search is explained sufficiently with `hibernate.search...`

**Review by kenfinnigan — APPROVED** [review]

**Review by gsmet — APPROVED** [review]
Given it's too late to change things in Micrometer, I think we will have to live with it.
